pkger/service_test.go

```diff
@@ -537,48 +532,104 @@ func TestService(t *testing.T) {
-						b.ID = influxdb.ID(id)
-						id++
-						return nil
-					}
-					fakeBktSVC.FindBucketByNameFn = func(_ context.Context, id influxdb.ID, s string) (*influxdb.Bucket, error) {
-						// forces the bucket to be created a new
-						return nil, errors.New("an error")
-					}
+			testLabelMappingFn := func(filename string, numExpected int, settersFn func() []ServiceSetterFn) func(t *testing.T) {
+				return func(t *testing.T) {
+					t.Helper()
+					testfileRunner(t, filename, func(t *testing.T, pkg *Pkg) {
+						fakeLabelSVC := mock.NewLabelService()
+						fakeLabelSVC.CreateLabelFn = func(_ context.Context, l *influxdb.Label) error {
+							l.ID = influxdb.ID(rand.Int())
+							return nil
+						}
+						numLabelMappings := 0
+						fakeLabelSVC.CreateLabelMappingFn = func(_ context.Context, mapping *influxdb.LabelMapping) error {
+							if mapping.ResourceID == 0 {
+								return errors.New("did not get a resource ID")
+							}
+							if mapping.ResourceType == "" {
+								return errors.New("did not get a resource type")
+							}
+							numLabelMappings++
+							return nil
+						}
+						svc := newTestService(append(settersFn(),
+							WithLabelSVC(fakeLabelSVC),
+							WithLogger(zaptest.NewLogger(t)),
+						)...)
 
-					fakeLabelSVC := mock.NewLabelService()
-					id = 1
-					fakeLabelSVC.CreateLabelFn = func(_ context.Context, l *influxdb.Label) error {
-						l.ID = influxdb.ID(id)
-						id++
-						return nil
-					}
-					numLabelMappings := 0
-					fakeLabelSVC.CreateLabelMappingFn = func(_ context.Context, mapping *influxdb.LabelMapping) error {
-						numLabelMappings++
-						return nil
-					}
-					fakeDashSVC := mock.NewDashboardService()
-					svc := newTestService(
-						zaptest.NewLogger(t),
-						WithBucketSVC(fakeBktSVC),
-						WithLabelSVC(fakeLabelSVC),
-						WithDashboardSVC(fakeDashSVC),
-					)
+						orgID := influxdb.ID(9000)
 
-					orgID := influxdb.ID(9000)
+						_, err := svc.Apply(context.TODO(), orgID, pkg)
+						require.NoError(t, err)
 
-					_, err := svc.Apply(context.TODO(), orgID, pkg)
-					require.NoError(t, err)
+						assert.Equal(t, numExpected, numLabelMappings)
+					})
+				}
+			}
 
-					assert.Equal(t, 4, numLabelMappings)
-				})
-			})
+			t.Run("successfully creates buckets with labels",
+				testLabelMappingFn(
+					"testdata/bucket_associates_label.yml",
+					4,
+					func() []ServiceSetterFn {
+						fakeBktSVC := mock.NewBucketService()
+						fakeBktSVC.CreateBucketFn = func(_ context.Context, b *influxdb.Bucket) error {
+							b.ID = influxdb.ID(rand.Int())
+							return nil
+						}
+						fakeBktSVC.FindBucketByNameFn = func(_ context.Context, id influxdb.ID, s string) (*influxdb.Bucket, error) {
+							// forces the bucket to be created a new
+							return nil, errors.New("an error")
+						}
+						return []ServiceSetterFn{WithBucketSVC(fakeBktSVC)}
+					},
+				),
+			)
+
+			t.Run("successfully creates dashboards with labels",
```

Comment:
putting together my checklist for adding resources, noticed I had holes in coverage for label mappings, this remedies that with the resusable test fn, now all resources have coverage (minus labels/packages b/c htey have no label associations)